### PR TITLE
fixes #847 - fall back to basic auth for pagerduty incidents api

### LINKED ISSUE #847: Re-instate pagerduty username and password for backwards compatibility in jsonapi
So that people's tooling doesn't break, but the data will not be used as the pagerduty gateway now uses the token auth

### PR DESCRIPTION
**Pull request by jessereynolds**
I haven't updated flapjack-diner yet so the pact specs will still need updating when that happens. But I think this should be OK to merge if all the tests pass.

### CODE REVIEW

**Comment by ali-graham** [comment]
LGTM.

File: spec/lib/flapjack/data/contact_spec.rb [hunk 1]
```diff
@@ -275,20 +277,26 @@
...
+                               'username'    => '',
+                               'password'    => ''})
   end
 
   it "sets pagerduty credentials for a contact" do
     contact = Flapjack::Data::Contact.find_by_id('c362', :redis => @redis)
     contact.set_pagerduty_credentials('service_key' => '567890123456789012345678',
                                       'subdomain'   => 'eggs',
-                                      'token'       => 'token123')
+                                      'token'       => 'token123',
```
Comment (Hobbsee) [inline spec/lib/flapjack/data/contact_spec.rb]: Both token and user/pass here?

File: spec/lib/flapjack/gateways/pagerduty_spec.rb [hunk 2]
```diff
@@ -72,10 +72,15 @@
        with(:headers => {'Authorization'=>'Token token=token123'}).
        to_return(:status => 200, :body => response.to_json, :headers => {})
 
+    stub_request(:get, "https://flpjck.pagerduty.com/api/v1/incidents?" +
```
Comment (Hobbsee) [inline spec/lib/flapjack/gateways/pagerduty_spec.rb]: Does this one actually get used?

**Comment by jessereynolds** [comment]
Thanks @ali-graham

File: spec/lib/flapjack/gateways/pagerduty_spec.rb (hunk 2 above)
Comment (jessereynolds) [inline spec/lib/flapjack/gateways/pagerduty_spec.rb]: Yes, there's two specs in there now that look for acknowledgements:
- "looks for acknowledgements via the PagerDuty API"
- "looks for acknowledgements via the PagerDuty API with basic auth"

The former tests token auth

File: spec/lib/flapjack/data/contact_spec.rb (hunk 1 above)
Comment (jessereynolds) [inline spec/lib/flapjack/data/contact_spec.rb]: Yes. The idea being you can set either token, or username and password, or both if you want to. The only logic in flapjack to select between one and the other is in the pagerduty gateway where it'll use token, or fall back to username and password. It seemed simplest to implement it this way. It means there should be no friction when either staying with your existing username+password auth, or switching to token (just add the token).

Comment (Hobbsee) [inline spec/lib/flapjack/data/contact_spec.rb]: That makes sense, good idea.

File: spec/lib/flapjack/gateways/pagerduty_spec.rb (hunk 2 above)
Comment (Hobbsee) [inline spec/lib/flapjack/gateways/pagerduty_spec.rb]: I thought line 116 was the stub for "looks for acknowledgements via the PagerDuty API with basic auth"

Comment (jessereynolds) [inline spec/lib/flapjack/gateways/pagerduty_spec.rb]: Oh right, I get you know. Thanks, I've removed that block in e28b77a

**Pull request merged**